include/seqan3/utility/bloom_filter/bloom_filter.hpp

```diff
@@ -0,0 +1,385 @@
+     * \param funs The number of hash functions. Default 2. At least 1, at most 5.
+     *
+     * \attention This constructor can only be used to construct **uncompressed** Bloom Filters.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/utility/bloom_filter/bloom_filter_constructor.cpp
+     */
+    bloom_filter(seqan3::bin_size size,
+                 seqan3::hash_function_count funs = seqan3::hash_function_count{2u})
+    //!\cond
+        requires (data_layout_mode == data_layout::uncompressed)
+    //!\endcond
+    {
+        size_in_bits = size.get();
+        hash_funs = funs.get();
+
+        if (hash_funs == 0 || hash_funs > 5)
+            throw std::logic_error{"The number of hash functions must be > 0 and <= 5."};
+        if (size_in_bits == 0)
+            throw std::logic_error{"The size of a bloom filter must be > 0."};
+
+        hash_shift = std::countl_zero(size_in_bits);
+        data = sdsl::bit_vector(size_in_bits);
+    }
+
+    /*!\brief Construct a compressed Bloom Filter.
+     * \param[in] bf The uncompressed seqan3::bloom_filter.
+     *
+     * \attention This constructor can only be used to construct **compressed** Bloom Filters.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/utility/bloom_filter/bloom_filter_constructor_compressed.cpp
+     */
+    bloom_filter(bloom_filter<data_layout::uncompressed> const & bf)
+    //!\cond
+        requires (data_layout_mode == data_layout::compressed)
+    //!\endcond
+    {
+        std::tie(size_in_bits, hash_shift, hash_funs) =
+            std::tie(bf.size_in_bits, bf.hash_shift, bf.hash_funs);
+
+        data = sdsl::sd_vector<>{bf.data};
+    }
+    //!\}
+
+    /*!\name Modifiers
+     * \{
+     */
+    /*!\brief Inserts a value into the bloom filter.
+     * \param[in] value The raw numeric value to process.
+     *
+     * \attention This function is only available for **uncompressed** Bloom Filters.
+     *
+     * \details
+     *
+     * ### Example
+     *
+     * \include test/snippet/utility/bloom_filter/bloom_filter_emplace.cpp
+     */
+    void emplace(size_t const value) noexcept
+    //!\cond
+        requires (data_layout_mode == data_layout::uncompressed)
+    //!\endcond
+    {
+        for (size_t i = 0; i < hash_funs; ++i)
+        {
+            size_t idx = hash_and_fit(value, hash_seeds[i]);
+            assert(idx < data.size());
+            data[idx] = 1;
+        };
+    }
+
+    /*!\brief Remove all values from the Bloom Filter by setting all bits to 0.
+     *
+     * \attention This function is only available for **uncompressed** Bloom Filters.
+     *
+     * \details
+     *
+     * While all values are removed from the vector, the size of the Bloom Filter is not changed.
+     * 
+     * ### Example
+     *
+     * \include test/snippet/utility/bloom_filter/bloom_filter_clear.cpp
+     */
+    void clear() noexcept
```

Comment:
We weren't sure about the `clear` name and here it becomes a bit more obvious that this might not be the best name - because `clear` usually means that `size == 0` afterwards.

Maybe `reset` or `erase` or something along those lines would be better?

This can also be follow-up since we would change the IBF too, but I don't think `clear` is a good name.

Reset would borrow from https://en.cppreference.com/w/cpp/utility/bitset/reset

Reply:
I think `reset` makes sense. If you agree, I can already change this.

Reply:
You can change it if you want to